x/distribution/client/cli/query.go

```diff
@@ -159,44 +135,86 @@ func GetCmdQueryDelegatorRewards(queryRoute string, cdc *codec.Codec) *cobra.Com
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
 
-			delegatorAddr, err := sdk.AccAddressFromBech32(args[0])
-			if err != nil {
-				return err
-			}
-
-			var (
-				route  string
-				params distr.QueryDelegationRewardsParams
-				result sdk.DecCoins
-			)
-
-			if len(args) == 1 {
-				// query for all rewards
-				params = distr.NewQueryDelegationRewardsParams(delegatorAddr, nil)
-				route = fmt.Sprintf("custom/%s/all_delegation_rewards", queryRoute)
-			} else {
+			valAddr := ""
+			if len(args) == 2 {
 				// query for rewards from a particular validator
-				validatorAddr, err := sdk.ValAddressFromBech32(args[1])
-				if err != nil {
-					return err
-				}
-
-				params = distr.NewQueryDelegationRewardsParams(delegatorAddr, validatorAddr)
-				route = fmt.Sprintf("custom/%s/delegation_rewards", queryRoute)
+				valAddr = args[1]
 			}
 
-			bz, err := cdc.MarshalJSON(params)
-			if err != nil {
-				return err
-			}
-
-			resp, err := cliCtx.QueryWithData(route, bz)
+			resp, err := QueryRewards(cliCtx, cdc, queryRoute, args[0], valAddr)
 			if err != nil {
 				return err
 			}
 
+			var result sdk.DecCoins
 			cdc.MustUnmarshalJSON(resp, &result)
 			return cliCtx.PrintOutput(result)
 		},
 	}
 }
+
+// QueryParams actually queries distribution params.
+func QueryParams(cliCtx context.CLIContext, queryRoute string) (PrettyParams, error) {
```

Comment:
Since this is used by both REST and CLI clients, it probably should not live under `cli/`.

Reply:
Created a client's subpackage: `common`. No alternatives to avoid import cycles.

Reply:
yeah that's precisely what I was thinking, a utils/core/common package 👍